src/component.js

```diff
@@ -0,0 +1,572 @@
+    r.depends = Component.getDependsList(herarchy, this.uuid).map( c => `${c.component.id} [${c.component.getUuid()}]`);
+
+
+    /*/
+    r.tags = [];
+    /*/
+    if (outputAsJson) {
+      cout(JSON.stringify(r, null, 2));
+    } else {
+      cout((require('yaml')).stringify(r));
+    }
+  }
+
+  /*
+  * 
+  * params:
+  */
+  async ls(cout, parents, depth, limit) {
+    this.logger.info(`ls ${this.uuid} - depth:'${depth}' limit:'${limit}'`);
+    await this.print(cout, parents, depth, limit);
+  }
+
+  /*
+  * Print hierarchy of components
+  * params:
+  */
+  async print(cout, parents, depth, limit, offset = '', last = false) {
+    let prefix = ' ';
+    if (parents && this.parent) {
+      offset = `${await this.parent.print(cout, parents, 0, 0)} └`;
+    }
+    // output yourself
+    let status = '';
+    if (!fs.existsSync(this.home)) {
+      status = '*'
+    }
+    const id = this.id === '' ? '/' : this.id;
+    cout(`${offset}${prefix}${id} ${status}`);
+    if (parents && offset.length) {
+      offset = offset.slice(0, -1) + ' ';
+    }
+    //
+    if (depth > 0) {
+      await this.buildAllChildren();
+      let more = 0;
+      let cnt = this.components.length;
+      if (limit > 0) {
+        if (limit < cnt) {
+          more = cnt - limit;
+          cnt = limit;
+        }
+      }
+      let no = offset;
+      if (offset.length && !parents) {
+        if (this.components.length) {
+          no = offset.substring(0, offset.length - 1) + '│';
+        }
+        if (last) {
+          no = offset.substring(0, offset.length - 1) + ' ';
+        }
+      }
+      for (const component of this.components) {
+        cnt--;
+        const delim = (cnt) ? (' ├') : (' └');
+        await component.print(cout, false, depth - 1, limit, `${no}${delim}`, cnt === 0);
+        if (!cnt) {
+          break;
+        }
+      }
+      if (more) {
+        cout(`${offset}   ... ${more} more`);
+      }
+    }
+    return offset;
+  }
+
+
+  //
+  async exec(recursive, filter, envFromCli, dryRun, command, input, _) {
+    this.logger.info(`exec ${this.uuid} - recursive:'${recursive}' command:'${command}' input:'${input}'`);
+    const herarchy = await this.unfoldHierarchy(this.uuid, this.id, this.home);
+    //
+    if (recursive) {
+      await this.buildAllChildren();
+      for (const component of this.components) {
+        await component.exec(recursive, filter, envFromCli, _, dryRun, command, input);
+      }
+    }
+    //
+    const script2Execute = scriptFactory.create(this.logger, 'clicmd', this.uuid, async (tln, script) => {
+      if (command) {
+        script.set([command])
+      } else if (input) {
+        script.set(input);
+      } else {
+        this.logger.warn(`${this.uuid} exec command input parameter is missing`);
+      }
+    });
+    const { scripts, env, dotenvs } = await this.collectScripts(herarchy, '', filter, envFromCli, _);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "scripts" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'scripts' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AUnusedVariable)